src/hotspot/share/gc/shenandoah/shenandoahBarrierSetNMethod.cpp

```diff
@@ -36,6 +36,13 @@
 #include "runtime/threadWXSetters.inline.hpp"
 
 bool ShenandoahBarrierSetNMethod::nmethod_entry_barrier(nmethod* nm) {
+
+  if (!is_armed(nm)) {
+    // Some other thread got here first and healed the oops
+    // and disarmed the nmethod.
```

Comment:
Suggestion for the comment (here and later):

```
    // Some other thread got here first and healed the oops
    // and disarmed the nmethod. No need to continue.
```

...and then later, under the lock:

```
    // Some other thread managed to complete while we were
    // waiting for lock. No need to continue.
```